libminifi/include/ThreadedSchedulingAgent.h

```diff
@@ -73,6 +73,8 @@ class ThreadedSchedulingAgent : public SchedulingAgent {
   ThreadedSchedulingAgent(const ThreadedSchedulingAgent &parent);
   ThreadedSchedulingAgent &operator=(const ThreadedSchedulingAgent &parent);
   std::shared_ptr<logging::Logger> logger_;
+
+  std::set<std::string> processors_running_;
```

Comment:
For sake of simplicity: the most straight forward container in case you have to search by value.
Performance doesn't matter here as this code piece is only executed during flow change (startup, shutdown, update). 
Because of the above I find this point of the guideline a bit overprotective. In case it's a core of a low latency software, yes, justify your container selection (and hopefully measure it, too!), but in our case, it's not that kind of software and definitely not a part of the code that has any effect on throughput. In this case I would suggest choosing the container that results in the most straight forward code.

Reply:
No offense, that's a good reason to use a different container. All I'm asking for is a code comment explaining the above for future readers.
e.g.: `// std::set for easy lookup/erase`

Reply:
Done